# The following patch works around a bug in PHP 5.  The shutdown
# procedure first unmapps the modules and then shuts down the
# evaluator which in turn calls the destructors for the objects in the
# objects_store and frees the cells bound to symbols.  Neither of
# these calls work when the module, which holds the classes for the
# objects to be destroyed, does not exist anymore.  To apply the patch
# cd to the directory that contains the PHP 5 source, for example 
# cd php-5.0.1 and then type: patch -p1 \
# <../php-java-bridge/php5-crash-in-evaluator-shutdown_workaround.patch

diff -c php-5.0.3RC1/main/main.c\~ php-5.0.3RC1/main/main.c
*** php-5.0.3RC1/main/main.c~	2004-10-18 12:36:35.000000000 +0200
--- php-5.0.3RC1/main/main.c	2004-12-05 13:16:30.690112425 +0100
***************
*** 1195,1200 ****
--- 1195,1202 ----
  	} zend_end_try();
  	
  	if (PG(modules_activated)) {
+ 		zend_objects_store_call_destructors(&EG(objects_store) TSRMLS_CC);
+ 		zend_hash_graceful_reverse_destroy(&EG(symbol_table));
  		zend_deactivate_modules(TSRMLS_C);
  	}
  

Diff finished at Sun Dec  5 13:16:43
